src/slot.rs

```diff
@@ -103,3 +104,15 @@ impl Add for Slot {
         }
     }
 }
+
+impl Slot {
+    pub fn num_hours(&self) -> usize {
+        let mut hours = 0;
+        let mut time = self.start;
+        while time < self.end {
+            time += Duration::hours(1);
+            hours += 1;
+        }
+        return hours;
```

Comment:
Why loop instead of end-start?

Reply:
Noted and agreed. Had overlooked the `.num_hours()` method available to the `Duration` struct. Will adjust in a new pr